website/blog/2024-11-04-test-smarter-not-harder.md

```diff
@@ -0,0 +1,161 @@
+
+### What does d@tA qUaL1Ty even mean?!
+
+High-quality data is *trusted* and *used frequently.* It doesn’t get argued over or endlessly scrutinized for matching to other data. Data *testing* should lead to higher data *quality* and insights, period.
+
+Best practices in data quality are still nascent. That said, a lot of important baseline work has been done here. There are [case](https://medium.com/@AtheonAnalytics/mastering-data-testing-with-dbt-part-1-689b2a025675) [studies](https://medium.com/@AtheonAnalytics/mastering-data-testing-with-dbt-part-2-c4031af3df18) on implementing dbt testing well. dbt Labs also has an [Advanced Testing](https://learn.getdbt.com/courses/advanced-testing) course, emphasizing that testing should spur action and be focused and informative enough to help address failures. You can even enforce testing best practices and dbt Labs’s own best practices using the [dbt_meta_testing](https://hub.getdbt.com/tnightengale/dbt_meta_testing/latest/) or [dbt_project_evaluator](https://github.com/dbt-labs/dbt-project-evaluator) packages and dbt Explorer’s [Recommendations](https://docs.getdbt.com/docs/collaborate/project-recommendations) page.
+
+The missing piece is still cohesion and guidance for everyday practitioners to help develop their testing framework.
+
+To recap, we’re going to start with:
+
+- identifying *data hygiene* issues
+- identifying *business-focused anomaly* issues
+- identifying *stats-focused anomaly* issues
+
+Next, we’ll prioritize. We will:
+
+- think through each concern in terms of the breadth of impact
+- decide if each concern should be at error or warning severity
+
+Get a pen and paper (or a google doc) and join us in constructing your own testing framework.
+
+## Identifying data quality issues in your pipeline
+
+Let’s start our framework by *identifying* types of data quality issues.
+
+In our daily work with customers, we find that data quality issues tend to fall into one of three broad buckets: *data hygiene, business-focused anomalies,* and *stats-focused anomalies.* Read the bucket descriptions below, and list 2-3 data quality concerns in your own business context that fall into each bucket.
+
+### Bucket One: Data Hygiene
+
+*Data hygiene* issues **are concerns you address in your [staging layer.](https://docs.getdbt.com/best-practices/how-we-structure/2-staging) Hygienic data meets your expectations around formatting, completeness, and granularity requirements. Here are a few examples.
+
+- *Granularity:* primary keys are unique and not null. Duplicates throw off calculations!
+- *Completeness:* columns that should always contain text, *do.* Incomplete data is less useful!
+- *Formatting:* email addresses always have a valid domain. Incorrect emails may affect marketing outreach!
```

Comment:
The examples here are good, I'm not convinced they need to end in !s. I also wonder if there's anything better than "less useful" for completeness?

Reply:
changed this in a fresh commit & expanded a bit on incomplete data usually needing to get thrown out & reducing overall analytical power